velox/vector/arrow/Bridge.cpp

```diff
@@ -135,4 +225,71 @@ void exportToArrow(const VectorPtr& vector, ArrowArray& arrowArray) {
   arrowArray.private_data = bridgeHolder.release();
 }
 
+void exportToArrow(const TypePtr& type, ArrowSchema& arrowSchema) {
+  arrowSchema.format = exportArrowFormatStr(type);
+  arrowSchema.name = nullptr;
+
+  // No additional metadata or support for dictionaries for now.
+  arrowSchema.metadata = nullptr;
+  arrowSchema.dictionary = nullptr;
+
+  // All supported types are semantically nullable.
+  arrowSchema.flags = ARROW_FLAG_NULLABLE;
+
+  // Allocate private data buffer holder and recurse down to children types.
+  auto bridgeHolder = std::make_unique<VeloxToArrowSchemaBridgeHolder>();
+  const size_t numChildren = type->size();
+
+  if (numChildren > 0) {
+    bridgeHolder->childrenRaw.resize(numChildren);
+    bridgeHolder->childrenOwned.resize(numChildren);
+
+    // If this is a RowType, hold the shared_ptr so we can set the
+    // ArrowSchema.name pointer to its internal `name` string.
+    if (type->kind() == TypeKind::ROW) {
+      bridgeHolder->rowType = std::dynamic_pointer_cast<const RowType>(type);
+    }
+
+    arrowSchema.children = &bridgeHolder->childrenRaw[0];
+    arrowSchema.n_children = numChildren;
+
+    for (size_t i = 0; i < numChildren; ++i) {
+      // Recurse down the children. We use the same trick of temporarily holding
+      // the buffer in a unique_ptr so it doesn't leak if the recursion throws.
+      //
+      // But this is more nuanced: for types with a list of children (like
```

Comment:
we also need to do it with `exportToArrow(const VectorPtr& vector, ArrowArray& arrowArray)` once it supports complex vectors, right?

Reply:
Likely, yes. I will check if there's a nice way to refactor this code when I get to that.